File: src/test/fixtures/csharp/CognitiveRules.cs
class CognitiveRules {
// straightLine: cognitive score 0.
static bool straightLine(bool ready, bool enabled, bool urgent) {
    return ready;
}

// booleanRun: cognitive score 1.
static bool booleanRun(bool ready, bool enabled, bool urgent) {
    return ready && enabled && urgent;
}

// mixedRuns: cognitive score 3.
static bool mixedRuns(bool ready, bool enabled, bool urgent) {
    return ready && enabled || urgent && ready;
}

// groupedRun: cognitive score 1.
static bool groupedRun(bool ready, bool enabled, bool urgent) {
    return ready && (enabled && urgent);
}

// negatedGroup: cognitive score 2.
static bool negatedGroup(bool ready, bool enabled, bool urgent) {
    return ready && !(enabled && urgent);
}

// plainElse: cognitive score 2.
static void plainElse(bool ready, bool enabled, bool urgent) {
    if (ready) { Deliver(); } else { Queue(); }
}

// flatChain: cognitive score 3.
static void flatChain(bool ready, bool enabled, bool urgent) {
    if (ready) { Deliver(); } else if (urgent) { Expedite(); } else { Queue(); }
}

// nestedChain: cognitive score 4.
static void nestedChain(bool ready, bool enabled, bool urgent) {
    if (enabled) { if (ready) { Deliver(); } else if (urgent) { Expedite(); } }
}

// nestedChecks: cognitive score 3.
static void nestedChecks(bool ready, bool enabled, bool urgent) {
    if (enabled) { if (ready) { Deliver(); } }
}

// dispatch: cognitive score 3.
static void dispatch(bool ready, bool enabled, bool urgent) {
    switch (ready) { case true: if (enabled) Deliver(); break; default: Queue(); break; }
}

// loopBody: cognitive score 3.
static void loopBody(bool ready, bool enabled, bool urgent) {
    while (ready) { if (enabled) Deliver(); }
}

// lambdaBody: cognitive score 2.
static void lambdaBody(bool ready, bool enabled, bool urgent) {
    System.Action action = () => { if (ready) Deliver(); }; action();
}

// emptyLambda: cognitive score 0.
static void emptyLambda(bool ready, bool enabled, bool urgent) {
    System.Action action = () => Deliver(); action();
}

// guardReturn: cognitive score 1.
static void guardReturn(bool ready, bool enabled, bool urgent) {
    if (!ready) return; Deliver();
}

// unbracedChecks: cognitive score 3.
static void unbracedChecks(bool ready, bool enabled, bool urgent) {
    if (enabled) if (ready) Deliver();
}

// doLoop: cognitive score 3.
static void doLoop(bool ready, bool enabled, bool urgent) {
    do { if (ready) Deliver(); } while (enabled);
}

// nestedFunction: cognitive score 2.
static void nestedFunction(bool ready, bool enabled, bool urgent) {
    Audit(); void Action() { if (ready) Deliver(); } Action();
}

// emptyFunction: cognitive score 0.
static void emptyFunction(bool ready, bool enabled, bool urgent) {
    Audit(); void Action() { Deliver(); } Action();
}

// cleanup: cognitive score 0.
static void cleanup(bool ready, bool enabled, bool urgent) {
    try { Deliver(); } finally { Cleanup(); }
}

// catches: cognitive score 4.
static void catches(bool ready, bool enabled, bool urgent) {
    try { Deliver(); } catch (TimeoutException error) { if (ready) Queue(); } catch (ArgumentException error) { Reject(); }
}

// labelJump: cognitive score 1.
static void labelJump(bool ready, bool enabled, bool urgent) {
    goto finished; finished: Deliver();
}

// ordinaryBreak: cognitive score 1.
static void ordinaryBreak(bool ready, bool enabled, bool urgent) {
    while (ready) { break; }
}

// bracedElseCheck: cognitive score 4.
static void bracedElseCheck(bool ready, bool enabled, bool urgent) {
    if (enabled) { Deliver(); } else { if (ready) { Deliver(); } }
}
}

File: src/test/fixtures/cpp/cognitive_rules.cpp
// straightLine: cognitive score 0.
auto straightLine(bool ready, bool enabled, bool urgent) {
    return ready;
}

// booleanRun: cognitive score 1.
auto booleanRun(bool ready, bool enabled, bool urgent) {
    return ready && enabled && urgent;
}

// mixedRuns: cognitive score 3.
auto mixedRuns(bool ready, bool enabled, bool urgent) {
    return ready && enabled || urgent && ready;
}

// groupedRun: cognitive score 1.
auto groupedRun(bool ready, bool enabled, bool urgent) {
    return ready && (enabled && urgent);
}

// negatedGroup: cognitive score 2.
auto negatedGroup(bool ready, bool enabled, bool urgent) {
    return ready && !(enabled && urgent);
}

// plainElse: cognitive score 2.
auto plainElse(bool ready, bool enabled, bool urgent) {
    if (ready) { deliver(); } else { queue(); }
}

// flatChain: cognitive score 3.
auto flatChain(bool ready, bool enabled, bool urgent) {
    if (ready) { deliver(); } else if (urgent) { expedite(); } else { queue(); }
}

// nestedChain: cognitive score 4.
auto nestedChain(bool ready, bool enabled, bool urgent) {
    if (enabled) { if (ready) { deliver(); } else if (urgent) { expedite(); } }
}

// nestedChecks: cognitive score 3.
auto nestedChecks(bool ready, bool enabled, bool urgent) {
    if (enabled) { if (ready) { deliver(); } }
}

// dispatch: cognitive score 3.
auto dispatch(bool ready, bool enabled, bool urgent) {
    switch (ready) { case true: if (enabled) deliver(); break; default: queue(); }
}

// loopBody: cognitive score 3.
auto loopBody(bool ready, bool enabled, bool urgent) {
    while (ready) { if (enabled) deliver(); }
}

// lambdaBody: cognitive score 2.
auto lambdaBody(bool ready, bool enabled, bool urgent) {
    auto action = [&]() { if (ready) deliver(); }; action();
}

// emptyLambda: cognitive score 0.
auto emptyLambda(bool ready, bool enabled, bool urgent) {
    auto action = [&]() { deliver(); }; action();
}

// guardReturn: cognitive score 1.
auto guardReturn(bool ready, bool enabled, bool urgent) {
    if (!ready) return; deliver();
}

// unbracedChecks: cognitive score 3.
auto unbracedChecks(bool ready, bool enabled, bool urgent) {
    if (enabled) if (ready) deliver();
}

// doLoop: cognitive score 3.
auto doLoop(bool ready, bool enabled, bool urgent) {
    do { if (ready) deliver(); } while (enabled);
}

// catches: cognitive score 4.
auto catches(bool ready, bool enabled, bool urgent) {
    try { deliver(); } catch (const TimeoutError& error) { if (ready) queue(); } catch (...) { reject(); }
}

// labelJump: cognitive score 1.
auto labelJump(bool ready, bool enabled, bool urgent) {
    goto finished; finished: deliver();
}

// ordinaryBreak: cognitive score 1.
auto ordinaryBreak(bool ready, bool enabled, bool urgent) {
    while (ready) { break; }
}

// bracedElseCheck: cognitive score 4.
auto bracedElseCheck(bool ready, bool enabled, bool urgent) {
    if (enabled) { deliver(); } else { if (ready) { deliver(); } }
}
